### PR TITLE
[WebAPI] Add 3 test cases for Presentation connection lifecycle

### PR DESCRIPTION
**Pull request by yhe39**
According the the comment, modify the test cases:
Use testharness assert to check several point in one test cases.
Remove the license information.
Delete the test cases that covered by idlharness.html.

Test reslults: under crosswalk for Android 20.50.532.0.
presentation/PresentationConnection_onclosed-manual.html  2/3 pass
presentation/PresentationConnection_onconnected-manual.html pending (the onconnect is not triggered)
presentation/PresentationConnection_onterminated-manual.html 1/3 pass

### CODE REVIEW

**Comment by yhe39** [comment]
@Honry @zqzhang PTAL. thanks a lot!

File: presentation/PresentationConnection_onclosed-manual.html [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  the connection state is set closed,<br/>
+  the onclose EventHandler is triggered.
+</p>
+<br/>
+<p>Click the button below to start the test.</p>
+<button id="presentStartBtn" onclick="startPresentation()">Start Presentation Test</button>
+
+<script>
+  setup({explicit_timeout: true});
+  var startBtn = document.getElementById("presentStartBtn");
```
Comment (zqzhang) [inline presentation/PresentationConnection_onclosed-manual.html]: This var is un-referenced. Then you don't need the `id` of the button.

File: presentation/PresentationConnection_onclosed-manual.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  var startPresentation = function () {
+    async_test(function(t) {
+      var url = "display/presentation.html";
```
Comment (zqzhang) [inline presentation/PresentationConnection_onclosed-manual.html]: Prefer a `support/` sub-directory for the presentation page.

File: presentation/PresentationConnection_onclosed-manual.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+          assert_true(connection instanceof PresentationConnection);
+          connection.onclose = t.step_func_done(function(evt) {
+            test(function() {
+              assert_equals(evt.type, "close");
+              assert_equals(connection.state, "closed");
+            }, 'the onclose is fired and the connection state is closed');
+          });
+          connection.close();
+          test(function() {
+            assert_equals(connection.state, "closed");
```
Comment (zqzhang) [inline presentation/PresentationConnection_onclosed-manual.html]: This test is wrong. See line 33. We have to check the closed status after we make sure it is really closed, that is the close event is fired.

File: presentation/PresentationConnection_onclosed-manual.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+<script>
+  setup({explicit_timeout: true});
+  var startBtn = document.getElementById("presentStartBtn");
+  var startPresentation = function () {
+    async_test(function(t) {
+      var url = "display/presentation.html";
+      var request = new PresentationRequest(url);
+      request.start()
+        .then(function(connection) {
+          assert_true(connection instanceof PresentationConnection);
```
Comment (zqzhang) [inline presentation/PresentationConnection_onclosed-manual.html]: Seems this assertion shall be in a test step, or you can make a promise test which wrappers test step in the test itself, see https://github.com/w3c/testharness.js/blob/master/testharness.js#L518

File: presentation/PresentationConnection_onclosed-manual.html [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+              assert_equals(connection.state, "closed");
+            }, 'the onclose is fired and the connection state is closed');
+          });
+          connection.close();
+          test(function() {
+            assert_equals(connection.state, "closed");
+          }, 'the connection state is closed after close');
+      })
+      .catch(function(ex) {
+        assert_unreached(ex.name + ":" + ex.message);
```
Comment (zqzhang) [inline presentation/PresentationConnection_onclosed-manual.html:36]: See comment on line 29.

File: presentation/PresentationConnection_onconnected-manual.html [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  setup({explicit_timeout: true});
+  var startBtn = document.getElementById("presentStartBtn");
+  var startPresentation = function () {
+    async_test(function(t) {
+      var url = "display/presentation.html";
+      var request = new PresentationRequest(url);
+      request.start()
+        .then(function(connection) {
+          assert_true(connection instanceof PresentationConnection);
+          connection.onconnect = t.step_func_done(function() {
```
Comment (zqzhang) [inline presentation/PresentationConnection_onconnected-manual.html:29]: If we are sure the `connect` event must be fired, we can test it together with `onclose` in sequence. See `EventWatcher` introduction at

https://github.com/w3c/testharness.js/blob/master/docs/api.md#promise-tests

File: presentation/PresentationConnection_onterminated-manual.html [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  setup({explicit_timeout: true});
+  var startBtn = document.getElementById("presentStartBtn");
+  var startPresentation = function () {
+    async_test(function(t) {
+      var url = "display/presentation.html";
+      var request = new PresentationRequest(url);
+      request.start()
+        .then(function(connection) {
+          assert_true(connection instanceof PresentationConnection);
+          connection.onterminate = t.step_func_done(function(evt) {
```
Comment (zqzhang) [inline presentation/PresentationConnection_onterminated-manual.html:29]: Same comment as for PresentationConnection_onconnected-manual.html

**Comment by yhe39** [comment]
@Honry @zqzhang PTAL. thanks a lot!

File: presentation/support/presentation.html [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  var addConnection = function(connection) {
+    connection.onconnected = function () {
+      this.onmessage = function (evt) {
+        this.send(evt.data);
+      };
+  };
+  navigator.receiver.connectionList
+    .then(function(list) {
+      list.onconnectionavailable = function(connections) {
+        addConnection(connections[connections.length-1]);
```
Comment (zqzhang) [inline presentation/support/presentation.html:17]: Put spaces before and after `-`. Others look good to me.

By the way, what's your testing result?

Comment (yhe39) [inline presentation/support/presentation.html:17]: test results:
PresentationConnection_onclosed-manual.html   pass 
PresentationConnection_onconnect-manual.html  fail (onconnect is not triggered)
PresentationConnection_onterminate-manual.html    fail (onterminate is not triggered but triggered the onclosed)

**Comment by yhe39** [comment]
Close the PR as mission complete :)

**Pull request closed without merging**